include/core/catchment/giuh/GIUH.hpp

```diff
@@ -0,0 +1,128 @@
+                std::vector<double> cdf_cumulative_freqs
+                )
+        {
+            this->catchment_id = std::move(catchment_id);
+            this->cdf_times = std::move(cdf_times);
+            // TODO: might be able to get this by calculating from times, rather than being passed
+            this->cdf_cumulative_freqs = std::move(cdf_cumulative_freqs);
+
+            carry_overs_list_head = nullptr;
+
+            // TODO: look at not hard-coding this later
+            cdf_regularity_seconds = 60;
+
+            // Interpolate regularized CDF (might should be done out of constructor, perhaps concurrently)
+            cdf_ordinate_times_seconds.push_back(0);
+            regularized_cdf_ordinates.push_back(0);
+            // Increment the ordinate time based on the regularity (loop below will do this at the end of each iter)
+            int time_for_ordinate = cdf_ordinate_times_seconds.back() + cdf_regularity_seconds;
+
+            // Loop through ordinate times, initializing all but the last ordinate
+            while (time_for_ordinate < this->cdf_times.back()) {
+                cdf_ordinate_times_seconds.push_back(time_for_ordinate);
+
+                // Find index 'i' of largest CDF time less than the time for the current ordinate
+                // Start by getting the index of the first time greater than time_for_ordinate
+                int cdf_times_index_for_iteration = 0;
+                while (this->cdf_times[cdf_times_index_for_iteration] < cdf_ordinate_times_seconds.back()) {
+                    cdf_times_index_for_iteration++;
+                }
+                // With the index of the first larger, back up one to get the last smaller
+                cdf_times_index_for_iteration--;
+
+                // Then apply equation from spreadsheet
+                double result = (time_for_ordinate - this->cdf_times[cdf_times_index_for_iteration]) /
+                                (this->cdf_times[cdf_times_index_for_iteration + 1] -
+                                 this->cdf_times[cdf_times_index_for_iteration]) *
+                                (this->cdf_cumulative_freqs[cdf_times_index_for_iteration + 1] -
+                                 this->cdf_cumulative_freqs[cdf_times_index_for_iteration]) +
+                                this->cdf_cumulative_freqs[cdf_times_index_for_iteration];
+                // Push that to the back of that collection
+                regularized_cdf_ordinates.push_back(result);
+
+                // Finally, increment the ordinate time based on the regularity
+                time_for_ordinate = cdf_ordinate_times_seconds.back() + cdf_regularity_seconds;
+            }
+
+            // Finally, the last ordinate time gets set to have everything
+            cdf_ordinate_times_seconds.push_back(time_for_ordinate);
+            regularized_cdf_ordinates.push_back(1.0);
+
+            // Also, now calculate the incremental values between each ordinate step
```

Comment:
Might be worth pulling the interpolation into a publicly exposed function so the GIUH could switch time bases dynamically.